interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java

```diff
@@ -628,6 +632,38 @@ public void gracefulShutdown() throws Exception {
     verifyNoMoreInteractions(responseObserver);
   }
 
+  /** Sends a large unary rpc with compute engine credentials. */
+  public void computeEngineCreds(String serviceAccount, String oauthScope) throws Exception {
+    ComputeEngineCredentials credentials = new ComputeEngineCredentials();
+    TestServiceGrpc.TestServiceBlockingStub stub = TestServiceGrpc.newBlockingStub(channel)
+        .configureNewStub()
+        .addInterceptor(new ClientAuthInterceptor(credentials, testServiceExecutor))
+        .build();
+    final SimpleRequest request = SimpleRequest.newBuilder()
+        .setFillUsername(true)
+        .setFillOauthScope(true)
+        .setResponseSize(314159)
+        .setResponseType(PayloadType.COMPRESSABLE)
+        .setPayload(Payload.newBuilder()
+            .setBody(ByteString.copyFrom(new byte[271828])))
+        .build();
+
+    final SimpleResponse response = stub.unaryCall(request);
+    assertEquals(serviceAccount, response.getUsername());
```

Comment:
The golden response is built with the username and oauthscope obtained from the received response, so:
1. we need that 3 asserts to check the username and oauthscope before checking the whole response.
2. the test will not fail when the received oauth scope is not exact same as the specified one.